core/src/main/java/org/neo4j/gds/core/concurrency/ParallelUtil.java

```diff
@@ -1487,8 +1489,19 @@ void pushBack(final T element) {
             pushedElement = element;
         }
     }
-
+    
     public static ForkJoinPool getFJPoolWithConcurrency(int concurrency) {
```

Comment:
If we do it as I sketched out above, we don't need the method to be thread-safe and just call it `createFJPool...`

Reply:
Got it!
Thanks for your review. I committed improvements based on all of your comments.